client/components/data/query-billing-transaction/index.jsx

```diff
@@ -0,0 +1,54 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import QueryBillingTransactions from '../query-billing-transactions';
+import { getPastBillingTransaction, isRequestingBillingTransaction } from 'state/selectors';
+import { requestBillingTransaction } from 'state/billing-transactions/individual-transactions/actions';
+
+class QueryBillingTransaction extends Component {
+	fetch( props ) {
+		const { transaction, transactionId, requestingBillingTransaction } = props;
+
+		if ( transaction || requestingBillingTransaction ) {
+			return;
+		}
+
+		props.requestBillingTransaction( transactionId );
+	}
+
+	componentDidMount() {
+		this.fetch( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		this.fetch( nextProps );
+	}
+
+	render() {
+		return <QueryBillingTransactions />;
```

Comment:
I would really like to avoid this - it breaks the convention of not returning anything in a query component.

But an even stronger argument for not doing this is - why should requesting a single transaction trigger a request to fetch all transactions? It would be the same to request all sites when a single site is being requested.